presto-feature-toggle/src/main/java/com/facebook/presto/features/binder/PrestoFeatureToggle.java

```diff
@@ -0,0 +1,144 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.features.binder;
+
+import com.facebook.presto.features.annotations.FeatureToggle;
+import com.facebook.presto.features.config.FeatureConfiguration;
+import com.facebook.presto.features.config.FeatureToggleConfiguration;
+import com.facebook.presto.features.strategy.FeatureToggleStrategy;
+import com.facebook.presto.features.strategy.FeatureToggleStrategyConfig;
+import com.facebook.presto.features.strategy.FeatureToggleStrategyFactory;
+import com.google.inject.Inject;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class PrestoFeatureToggle
+{
+    private final Map<String, Feature<?>> featureMap;
+    private final Map<String, Object> featureInstanceMap;
+    private final FeatureToggleStrategyFactory featureToggleStrategyFactory;
+    private final FeatureToggleConfiguration featureToggleConfiguration;
+
+    @Inject
+    public PrestoFeatureToggle(
+            @FeatureToggle("feature-map") Map<String, Feature<?>> featureMap,
+            @FeatureToggle("feature-instance-map") Map<String, Object> featureInstanceMap,
+            FeatureToggleStrategyFactory featureToggleStrategyFactory,
+            FeatureToggleConfiguration featureToggleConfiguration)
+    {
+        this.featureMap = featureMap;
+        this.featureInstanceMap = featureInstanceMap;
+        this.featureToggleStrategyFactory = featureToggleStrategyFactory;
+        this.featureToggleConfiguration = featureToggleConfiguration;
+        this.featureMap.values().forEach(f -> f.setContext(this));
+    }
+
+    public boolean isEnabled(String featureId)
+    {
+        Feature<?> feature = featureMap.get(featureId);
+        FeatureConfiguration featureConfiguration = feature.getConfiguration();
+        AtomicBoolean enabled = new AtomicBoolean(true);
+        FeatureConfiguration configuration = featureToggleConfiguration.getFeatureConfiguration(featureId);
+        if (configuration != null) {
+            if (!configuration.isEnabled()) {
+                return false;
+            }
+            enabled.set(configuration.isEnabled());
+            if (configuration.getFeatureToggleStrategyConfig().isPresent()) {
+                String toggleStrategyClass = configuration.getFeatureToggleStrategyConfig().get().getToggleStrategyName();
+                if (toggleStrategyClass != null) {
+                    Optional<FeatureToggleStrategy> strategy = Optional.ofNullable(featureToggleStrategyFactory.get(toggleStrategyClass));
+                    return strategy
+                            .map(featureToggleStrategy -> enabled.get() && featureToggleStrategy.check(configuration))
+                            .orElse(enabled.get());
+                }
```

Comment:
Just do a null check instead of wrapping in Optional.  Please apply this advice to the entire file.